tripy/tests/integration/test_flatten.py

```diff
@@ -55,3 +57,15 @@ def test_flatten_with_unknown_dims(self, eager_or_compiled):
         a = tp.ones((2, 3, 4, 5))
         b = eager_or_compiled(tp.flatten, a, start_dim=1, end_dim=-1)
         assert np.array_equal(cp.from_dlpack(b).get(), np.ones((2, 60), dtype=np.float32))
+
+    @pytest.mark.parametrize(
+        "shape, start_dim, end_dim, expected_shape",
+        test_cases,
+    )
+    def test_flatten_tensor_method(self, shape, start_dim, end_dim, expected_shape, eager_or_compiled):
+        """Test that tensor.flatten() method works and produces same result as free function."""
+        cp_a = cp.arange(np.prod(shape)).reshape(shape).astype(np.float32)
+        a = tp.Tensor(cp_a)
+        b = eager_or_compiled(lambda t: t.flatten(start_dim=start_dim, end_dim=end_dim), a)
```

Comment:
I think we could just parametrize these tests on arguments to `eager_or_compiled`?